src/core_test/mpas_test_core_halo_exch.F

```diff
@@ -1370,5 +1380,167 @@ subroutine test_core_halo_exch_validate_fields(domain, threadErrs, err)!{{{
+      call MPAS_pool_get_array(meshPool, "nEdgesOnCell", nEdgesOnCell)
+      call mpas_log_write(' halo_adj_ cellsOnCell size:$ix$i', intArgs=(/size(cellsOnCell,2), size(cellsOnCell, 1)/))
+
+      interiorCells = findInteriorCells(nCellsSolve, cellsOnCell, nEdgesOnCell)
+
+      ! run the adjoint halo, this will update owned cells
+      call mpas_dmpar_exch_halo_adj_field(real2DField, interiorCells=interiorCells)
+      !call mpas_dmpar_exch_halo_adj_field(real2DField)
+
+      !do while ( associated(block) )
+
+         call mpas_log_write(' test_halo_adj_exch_fields block:$i next is null:$l', &
+            intArgs=(/block%blockID/), logicArgs=(/.not. associated(block%next)/))
+
+         ! get the real2D array after calling mpas_dmpar_exch_halo_adj_field
+         call mpas_pool_get_array(haloExchTestPool, 'cellPersistReal2D', real2D)
+         call mpas_log_write(' halo orig size($i, $i) adj size($i, $i)', &
+                         intArgs=(/size(real2Dorig, 2), size(real2Dorig, 1), &
+                            size(real2D, 2), size(real2D, 1)/))
+
+         ! check the adjoint halo operation populated fields correctly
+         err = check_adjoint_values(nCellsSolve, real2Dorig, real2D, interiorCells)
+         !block => block % next
+      !end do
+   end subroutine test_halo_adj_exch_fields
+
+   !***********************************************************************
+   !> \brief   MPAS Test check pre & post adjoint exchange values
+   !> \details 
+   !>  This routine checks the pre-adjoint halo exchange values aganst
+   !>  post-adjoint halo exhange values.
+   !>  Interior cell's values aren't expected to change, and border cell's values are 
+   ! > expected to change.
+   !>  Returns 0 on success, non-0 on failure.
+   !-----------------------------------------------------------------------
+   integer function check_adjoint_values(nCellsSolve, orig, adjoint, interiorCells) 
+      integer, pointer, intent(in) :: nCellsSolve !< the number of local owned cells
+      real (kind=RKIND), dimension(:,:), intent(in) :: orig !< values of the cells before applying the adjoint exchange
+      real (kind=RKIND), dimension(:,:), intent(in) :: adjoint !< values of cells after applying the adjoint exchange
+      integer, dimension(:), intent(in) :: interiorCells !< array indicating a cell is interior on on the edge
+
+      integer :: i, j, iCell, iEdgeOnCell, nError, nInterior, nEdge
+      integer :: iDim1, iDim2
+
+      nError = 0;
+      iDim1 = nCellsSolve
+      iDim2 = size(orig, dim=1)
+      nInterior = 0
+      nEdge = 0
+
+      do i = 1, iDim1
+         do j = 1, iDim2
+            if (interiorCells(i) .eq. 1) then
+               if (j == 1) then
+                  nInterior = nInterior + 1
+                  ! interior cells shouldn't have changed
+                  if (orig(j, i) /= adjoint(j, i)) then
+                     call mpas_log_write(' halo changed value for interior cell at:$i:$i orig:$r new:$r', &
```

Comment:
This check showed around 150 interior cells were updated via the halo adjoint exchange.